rspace/src/main/scala/coop/rchain/rspace/history/Trie.scala

```diff
@@ -39,8 +62,29 @@ object Trie {
       }((codecByteVector :: codecNonEmptyPointer).as[Skip])
 
   def hash[K, V](trie: Trie[K, V])(implicit codecK: Codec[K], codecV: Codec[V]): Blake2b256Hash =
-    codecTrie[K, V]
-      .encode(trie)
-      .map((vector: BitVector) => Blake2b256Hash.create(vector.toByteArray))
-      .get
+    if(TrieCache.useCache) {
+      trie.asInstanceOf[HashedEntity].initialize {
```

Comment:
Do you intend to leave this `asInstanceOf` here?

Reply:
No,  `asInstanceOf` is already removed, and `if(TrieCache.useCache)` should be removed before final merging.